src/murmur/Cert.cpp

```diff
@@ -132,57 +234,62 @@ void Server::initializeCert() {
 	// generate a new one below.
 	if (issuer == QString::fromUtf8("Murmur Autogenerated Certificate")) {
 		log("Old autogenerated certificate is unusable for registration, invalidating it");
-		qscCert = QSslCertificate();
-		qskKey  = QSslKey();
+		qlCertificateChain.clear();
+		qlCertificateChain << QSslCertificate();
+		qskKey = QSslKey();
 	}
 
 	// If we have a cert, and it's a self-signed one, but we're binding to
 	// all the same addresses as the Meta server is, use it's cert instead.
 	// This allows a self-signed certificate generated by Murmur to be
 	// replaced by a CA-signed certificate in the .ini file.
-	if (!qscCert.isNull() && issuer.startsWith(QString::fromUtf8("Murmur Autogenerated Certificate"))
-		&& !Meta::mp.qscCert.isNull() && !Meta::mp.qskKey.isNull() && (Meta::mp.qlBind == qlBind)) {
-		qscCert         = Meta::mp.qscCert;
-		qskKey          = Meta::mp.qskKey;
-		qlIntermediates = Meta::mp.qlIntermediates;
-
-		if (!qscCert.isNull() && !qskKey.isNull()) {
+	if (!qlCertificateChain.isEmpty() && !qlCertificateChain[0].isNull()
+		&& issuer.startsWith(QString::fromUtf8("Murmur Autogenerated Certificate"))
+		&& !Meta::mp.qlCertificateChain.isEmpty() && !Meta::mp.qskKey.isNull() && (Meta::mp.qlBind == qlBind)) {
+		qlCertificateChain.clear();
+		qlCertificateChain = Meta::mp.qlCertificateChain;
+		qskKey             = Meta::mp.qskKey;
+		if (!qlCertificateChain.isEmpty() && !qlCertificateChain[0].isNull() && !qskKey.isNull()) {
 			bUsingMetaCert = true;
 		}
 	}
 
 	// If we still don't have a certificate by now, try to load the one from Meta
-	if (qscCert.isNull() || qskKey.isNull()) {
+	if (qlCertificateChain.isEmpty() || qlCertificateChain[0].isNull() || qskKey.isNull()) {
 		if (!key.isEmpty() || !crt.isEmpty()) {
 			log("Certificate specified, but failed to load.");
 		}
 
-		qskKey          = Meta::mp.qskKey;
-		qscCert         = Meta::mp.qscCert;
-		qlIntermediates = Meta::mp.qlIntermediates;
+		qlCertificateChain.clear();
+		qlCertificateChain = Meta::mp.qlCertificateChain;
+		qskKey             = Meta::mp.qskKey;
 
-		if (!qscCert.isNull() && !qskKey.isNull()) {
+		if (!qlCertificateChain.isEmpty() && !qlCertificateChain[0].isNull() && !qskKey.isNull()) {
 			bUsingMetaCert = true;
 		}
 
 		// If loading from Meta doesn't work, build+sign a new one
-		if (qscCert.isNull() || qskKey.isNull()) {
+		if (qlCertificateChain.isEmpty() || qlCertificateChain[0].isNull() || qskKey.isNull()) {
 			log("Generating new server certificate.");
-
-			if (!SelfSignedCertificate::generateMurmurV2Certificate(qscCert, qskKey)) {
+			if (qlCertificateChain.isEmpty()) {
+				qlCertificateChain << QSslCertificate();
+			}
+			if (!SelfSignedCertificate::generateMurmurV2Certificate(qlCertificateChain[0], qskKey)) {
 				log("Certificate or key generation failed");
 			}
 
-			setConf("certificate", qscCert.toPem());
+			setConf("certificate", chainToPem(qlCertificateChain));
```

Comment:
not sure if it is a good idea to reuse the same config-name for a different thing. Maybe it would be better if we used `certificate-chain` as the config key